src/service/ApiContainer.ts

```diff
@@ -56,4 +61,32 @@ export class ApiContainer {
   private getLogger(): Logger {
     return pino({ prettyPrint: { translateTime: true } });
   }
+
+  private async getAuthenticationMiddleware(): Promise<BasicAuthenticationMiddleware> {
+    const db = await this.getDatabase();
+    const repository = new OrganisationAuthenticationRepository(db);
+    const index = await repository.getPasswordIndex();
+
+    return new BasicAuthenticationMiddleware(index, this.getCryptography());
+  }
+
+  @memoize
```

Comment:
is this annotation specific to a framework, dont remember node having this.

Reply:
it is - it's one I made actually:

https://github.com/linusnorton/memoized-decorator